### PR TITLE
If capabilities are explicitly added then also add them to the ambien…

### PR DESCRIPTION
**Pull request by vinayakankugoyal**
…t set.

#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
his change makes it so that we add capabilities to the ambient set if they are explicitly added. So in kubernetes if we add a capability to the container securityContext then that capability will be added to the ambient set in addition to the inheritable, effective, permitted and bounding sets. This will allow users to run a container as non-root while giving it just the capabilities it needs, without having to apply file capabilities and allowing privilege escalation.

If ALL capabilities are added then we don't add ALL capabilities to the ambient set because that would make a non-root user essentially root. We also don't add the defaultUnixCaps i.e. the default capabilities that are always added to inheritable, effective, permitted and bounded sets to the ambient capability set as that will also make non-root user as power full as root.

#### Does this PR introduce a user-facing change?

```release-note
Explicitly added capabilities are added to the ambient set.
```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
@vinayakankugoyal thanks for the PR but the CI requires PR's to be signed.  Please `git commit --amend -s`, fetch, rebase and push.

**Comment by vinayakankugoyal** [comment]
@mrunalp - this is RE: kubernetes/enhancements#2757

**Comment by haircommander** [comment]
/ok-to-test

full review will come later, but we can run CI for now, thanks for opening @vinayakankugoyal

File: server/container_create.go [hunk 1]
```diff
@@ -351,6 +353,30 @@ func setupCapabilities(specgen *generate.Generator, caps *types.Capability, defa
 		}
 	}
 
+	if addDefaultCapbilities {
```
Comment (saschagrunert) [inline server/container_create.go:356]:
```suggestion
	if !addAll && !dropAll {
```

to make `addDefaultCapbilities` obsolete

**Comment by mrunalp** [comment]
@vinayakankugoyal Thanks for the PR!
What about the case where a user drops ALL and adds back the default set?

**Comment by vinayakankugoyal** [comment]
All those would be added to to the ambient set. But that in that case the user is choosing to do so, by explicitly adding them. We should be secure by default and not add anything to the ambient set if nothing is explicitly added is a safe starting point.

**Comment by haircommander** [comment]
I think this PR doesn't make sense right now. I think we'd want an alternate way for container's to specify ambient caps, ideally with proper pod API support in kube. I am closing this. Thanks for your PR @vinayakankugoyal

**Pull request closed without merging**